### PR TITLE
add com.github.Murmele.Gittyup

### PR DESCRIPTION
**Pull request by Murmele**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
and lease indent the manifest file properly

**Comment by nedrichards** [comment]
bot, build io.github.gitahead.gitahead

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

File: com.github.Murmele.Gittyup.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        /* needed to start external applications like kdiff3 (merge tool) ... */
+        "--filesystem=/usr/bin",
+        "--filesystem=/bin",
+        /*we use the keyring to store credentials*/
+        "--filesystem=xdg-run/keyring",
+        /*for git repos that require ssh keys*/
+        "--socket=ssh-auth",
+        "--talk-name=org.freedesktop.secrets",
+        "--talk-name=org.freedesktop.Notifications",
+        "--env=PATH=:/usr/bin:/app/bin:/app/bin/x86_64-linux:/app/bin/aarch64-linux"
```
Comment (hfiguiere) [inline com.github.Murmele.Gittyup.json:22]: I don't see why this is necessary

File: com.github.Murmele.Gittyup.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    ],
+    "build-options":
+     {
+        "env": {
+            "PATH": "$PATH:/usr/bin:/app/bin"
```
Comment (hfiguiere) [inline com.github.Murmele.Gittyup.json:27]: this is already set this way. It's not necessary.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

File: com.github.Murmele.Gittyup.json (hunk 2 above)
Comment (Murmele) [reply com.github.Murmele.Gittyup.json]: thank you

File: com.github.Murmele.Gittyup.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "sdk": "org.kde.Sdk",
+    "command": "Gittyup",
+    "finish-args": [
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--share=network",
+        "--filesystem=home",
+        /* needed to start external applications like kdiff3 (merge tool) ... */
+        "--filesystem=/usr/bin",
+        "--filesystem=/bin",
```
[the remark this answers is not in the document]
Comment (Murmele) [reply com.github.Murmele.Gittyup.json]: @barthalion I don't exactly understand what you mean. Gittyup searches through the search paths, which external application is installed and if yes it executes it with arguments. But if I don't add the paths, I'm not able to find the applications.

So my new idea. Check all compatible applications with flatpak-spawn --host which <app> to find out if the application exists or not?

**Comment by barthalion** [comment]
No complaints from me, apart from the commit thing in the manifest. Let us know when you consider it ready to merge.

**Comment by hfiguiere** [comment]
> No complaints from me, apart from the commit thing in the manifest. Let us know when you consider it ready to merge.

It still builds with a `branch` instead of a `commit`.

File: com.github.Murmele.Gittyup.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                "install -Dm755 ../rsrc/Gittyup.iconset/icon_256x256.png /app/share/icons/hicolor/256x256/apps/gitahead.png",
+                "install -Dm755 ../rsrc/Gittyup.iconset/icon_256x256@2x.png /app/share/icons/hicolor/256x256@2x/apps/gitahead.png",
+                "install -Dm755 ../rsrc/Gittyup.iconset/icon_512x512.png /app/share/icons/hicolor/512x512/apps/gitahead.png"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/Murmele/Gittyup",
+                    "branch": "master"
+                    /*"commit": "5683fa894b2b3b8dbd4421ca8ffd67138dd2d37e"*/
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.github.Murmele.Gittyup.json:47]: maybe if can be fixed now, can't it?

**Comment by Murmele** [comment]
> > No complaints from me, apart from the commit thing in the manifest. Let us know when you consider it ready to merge.
> 
> It still builds with a `branch` instead of a `commit`.

Yes I would like to fix the problem with external applications first.

**Comment by Murmele** [comment]
@hfiguiere Now I'm ready for now

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

File: com.github.Murmele.Gittyup.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (Murmele) [reply com.github.Murmele.Gittyup.json]: It is fixed

File: com.github.Murmele.Gittyup.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--share=network",
+        "--share=ipc",
+        "--filesystem=home",
+        "--filesystem=/tmp", /* Needed to store temporary files, which should be opened by external (non sandboxed) applications like kdiff3. Did not find another way to share a temporary file to the host (only creating a tmp folder in the home folder, I don't like) */
+        /*we use the keyring to store credentials*/
+        "--filesystem=xdg-run/keyring",
+        /*for git repos that require ssh keys*/
+        "--socket=ssh-auth",
```
Comment (nedrichards) [inline com.github.Murmele.Gittyup.json:17]: can you add `--device=dri`? Otherwise it complains about the lack of access to the gpu on wayland.

Comment (Murmele) [reply com.github.Murmele.Gittyup.json:17]: yes I can add

**Comment by Murmele** [comment]
bot, build com.github.Murmele.Gittyup

**Comment by barthalion** [comment]
bot, build com.github.Murmele.Gittyup

**Comment by Murmele** [comment]
@barthalion Do you have any insecurities?

**Comment by Murmele** [comment]
@barthalion  when it will be available on the flathub page?

**Comment by hfiguiere** [comment]
give it a few hours.

**Comment by Murmele** [comment]
Ok thanks :)

**Pull request closed without merging**